sdk/cosmosdb/cosmos/src/client/ChangeFeed/ChangeFeedForPartitionKey.ts

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { InternalChangeFeedIteratorOptions } from "./InternalChangeFeedOptions";
+import { ChangeFeedIteratorResponse } from "./ChangeFeedIteratorResponse";
+import { Resource } from "../../client";
+import { ClientContext } from "../../ClientContext";
+import { Constants, ResourceType } from "../../common";
+import { FeedOptions, Response, ErrorResponse } from "../../request";
+import { ContinuationTokenForPartitionKey } from "./ContinuationTokenForPartitionKey";
+import { ChangeFeedIteratorV2 } from "./ChangeFeedIteratorV2";
+import { PartitionKey } from "../../documents";
+import { PartitionKeyRange } from "../../client";
+import { IEpkRange } from "./IEpkRange";
+/**
+ * @hidden
+ * Provides iterator for change feed.
+ *
+ * Use `Items.getChangeFeedIterator()` to get an instance of the iterator.
+ */
+export class ChangeFeedForPartitionKey<T> extends ChangeFeedIteratorV2<T> {
+  private continuationToken: ContinuationTokenForPartitionKey;
+  private startTime: string;
+
+  /**
+   * @internal
+   */
+  constructor(
+    private clientContext: ClientContext,
+    private resourceId: string,
+    private resourceLink: string,
+    private rId: string,
+    private partitionKey: PartitionKey,
```

Comment:
One important caveat: Cosmos DB recently added Hierarchical Partition Keys. In this case, a Partition Key might not represent a single value but an EPKRange. For example, creating a container partitioned by [`/country`, `/city`, `/neighborhood`] which is a hierarchical partition definition and then creating a Change Feed (or even Query) for PartitionKey = ['USA', 'Seattle'] is not 1 value but rather includes all the PartitionKeys that are neighborhoods in Seattle which is actually an EPK Range. I am not sure if NodeJS currently has support for Hierarchical Partition Keys but if it does, it's worth making sure that in that case, it goes to the other iterator type, not this one.

Reply:
Thanks for pointing this out! This case is not covered right now. We will specifically mention this in the documentation.